nlp/2021_06_29_long_range_transformers/LongRangeTransformers.ipynb

```diff
@@ -0,0 +1,1940 @@
+    "\n",
+    "Let's put these architectures to the test and see which one comes out on top 🏆!  \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "l8ymZTD_BSZS"
+   },
+   "source": [
+    "## 🛠️ Getting started: Install packages & download models\n",
+    "\n",
+    "The below cells will setup everything that is required to get started with keyword extraction:\n",
+    "\n",
+    "* Install python specific packages\n",
+    "* Import required packages"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {
+    "id": "eDI558ScBWpZ"
+   },
+   "outputs": [],
+   "source": [
+    "!pip install -q sklearn transformers datasets torch plotly"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "id": "x7x9s2l_VMLV"
+   },
+   "outputs": [],
+   "source": [
+    "import time \n",
+    "import sys \n",
+    "import json\n",
+    "import pandas as pd\n",
+    "from enum import Enum\n",
+    "import torch\n",
+    "\n",
+    "import plotly.express as px\n",
+    "import plotly.graph_objects as go\n",
+    "\n",
+    "\n",
+    "from sklearn.metrics import accuracy_score, precision_recall_fscore_support\n",
+    "from transformers import BigBirdTokenizerFast, BigBirdForSequenceClassification, RobertaTokenizer, RobertaForSequenceClassification, LongformerForSequenceClassification, TrainingArguments, Trainer, LongformerTokenizerFast\n",
+    "from datasets import load_dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "UOdkK5pXBSZT"
+   },
+   "source": [
+    "## 💾 Dataset & downstream task\n",
+    "\n",
+    "We will use the [IMDB](https://huggingface.co/datasets/imdb) movie review dataset for binary sentiment classification. In the paper publication of LongFormer and BigBird, both architectures were compared against RoBERTa with this exact dataset.\n",
+    "*   13.6 % of the reviews are larger than 512 wordpieces \n",
+    "*   evaluation was done on the entire dataset and the performance improvement was .4% and .8% compared to the RoBERTa architecture for LongFormer and BigBird respectively 💪.\n",
+    "\n",
+    "We aim to gain more insight in when to use which architecture, therefore we will go *one step beyond 🔥*, and evaluate the architectures on 2 distinct subsets of the data, namely: \n",
+    "\n",
+    "1.   Short sequences: containing less than 512 wordpieces but more than 256 wordpieces\n",
+    "2.   Long sequences: containing more than 512 wordpieces\n",
+    "\n",
+    "\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "s63EKk7JVdsY",
+    "outputId": "5d2a0c41-0507-4f4d-b9be-1ae24019b74f"
+   },
+   "outputs": [],
+   "source": [
+    "ds = load_dataset('imdb')\n",
+    "\n",
+    "class IMDBDatasetTokenCountType(Enum):\n",
```

Comment:
what is the purpose of this class? I could be wrong, but it feels a bit like unnecessary overhead

Reply:
We discussed this shortly in the call. Initially, it was to have a cleaner implementation for post-analysis (plotting, ...), but I agree this now seems a bit overkill since I am only using 2 enum types, I think a dictionary of